harmonica/forward/tesseroid.py

```diff
@@ -658,10 +653,18 @@ def _longitude_continuity(tesseroids):
         The array must have the following shape: (``n_tesseroids``, 6), where
         ``n_tesseroids`` is the total number of tesseroids.
         All tesseroids must have valid boundary coordinates.
+
+    Returns
+    -------
+    tesseroids : 2d-array
+        Modified boundaries of the tesseroids.
     """
+    # Copy the tesseroids in order to avoid modifying the original tesseroids array
+    tesseroids = tesseroids.copy()
     west, east = tuple(tesseroids[:, i] for i in range(2))
     west %= 360
     east %= 360
     tess_to_be_changed = west > east
```

Comment:
I would put these two lines below and the copy in a `if np.any(tess_to_be_changed)` block to avoid copying if we don't need to.

Reply:
You're absolutely right!

Reply:
**Update**: I don't think we can copy only the west and east boundaries, because when we need to incorporate the modified east and west into the tesseroids array, a completely new tesseroids array must be created.

Reply:
>  I don't think we can copy only the west and east boundaries

I agree. But if `np.any(tess_to_be_changed) == False` then we don't need to copy `tesseroids` at all and could return the original tesseroids instead. Right?

Reply:
In that sense, do lines 665 and 666 (:metal:) need to be inplace operations?

Reply:
Each time I go back with this longitude continuity functions I have to rethink everything (I'm thinking to write a little document to explain it to myself).

Lines 665 and 666 move every longitude boundary to the [0, 360) interval. So, if we passed a tesseroid like (350, 370, ...), it will move it to (350, 10, ...). Then the `tess_to_be_changed` will  mark it as needed to be changed, moving it to the [-180, 180) interval: (-10, 10, ...).

So, lines 665 and 666 must be run independently if no tesseroid should be moved to [-180, 180) afterwards.

Maybe we can check if `west` and `east` fall either on [0, 360) or [-180, 180) and if `west < east`. If any of these fail, then we should perform the transformation of the tesseroids, i.e. copying it. That's the only option I can think to avoid copying the entire tesseroids array every time we run the forward model independently of the boundaries we pass.

Reply:
What if we run `_check_tesserroids` first and assume that all are either [0, 360] or [-180, 180]? That seems reasonable to me. Allowing `(350, 370, ...)` feels wrong. If that's the case, then we can avoid the in place `%` operation (and the copy it requires).

Reply:
Ok, I've updated the standards for what we consider a valid tesseroid.

A valid tesseroid should have:
- latitudinal boundaries within the [-90, 90] degrees interval,
- north boundaries greater or equal than the south boundaries,
- radial boundaries positive or zero,
- top boundaries greater or equal than the bottom boundaries,
- longitudinal boundaries within the [-180, 360] degrees interval,
- longitudinal interval must not be greater than one turn around the globe.

Some valid tesseroids have its west boundary greater than the east one, e.g.
``(350, 10, ...)``. On these cases the ``_longitude_continuity`` function is applied
in order to move the longitudinal coordinates to the [-180, 180) interval. Any valid
tesseroid should have east boundaries greater than the west boundaries before or
after applying longitude continuity.

Reply:
:+1: looks good to me! Should the `_longitude_continuity` function be deleted now?

Reply:
Nope. I call it from the `_check_tesseroids` function. I decided to leave it on a separate function because `_check_tesseroids` is already very cluttered.

Reply:
:+1: right, I missed that. Carry on :)